snorkel/augmentation/policy/core.py

```diff
@@ -0,0 +1,31 @@
+import random
+from typing import List
+
+from snorkel.augmentation.tf import TransformationFunction, TransformationFunctionMode
+from snorkel.types import DataPoint
+
+
+class AugmentationPolicy:
+    def __init__(self, tfs: List[TransformationFunction]):
+        self._tfs = tfs
+
+    def set_tf_mode(self, mode: TransformationFunctionMode) -> None:
+        for tf in self._tfs:
+            tf.set_mode(mode)
+
+    def apply(self, x: DataPoint) -> DataPoint:
+        raise NotImplementedError
```

Comment:
Okay how I was originally thinking about this, and based on what "policy" usually means: thoughts on having the `AugmentationPolicy` class just yield an ordering / generator of TF ids to be applied, but leave the actual application of TFs to the TF applier (so e.g. it can be optimized, and factored out better)?

Reply:
Sounds good